agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFile.java

```diff
@@ -109,6 +117,8 @@
 
     logDir = initLogDir();
     directory = logDir + STATUS_FILE_DIRECTORY;
+    shouldWrite =
+        DiagnosticsHelper.useAppSvcRpIntegrationLogging() && new File(directory).canWrite();
```

Comment:
I'm thinking now this may be too early to call `DiagnosticsHelper.useAppSvcRpIntegrationLogging()`, since that relies method relies on setAgentJarFile() which isn't called until MainEntryPoint.

Which is likely why it was in `shouldWrite()` method previously.

Let's cache `isReadOnly` here based on `new File(directory).canWrite()`, and then go back to `shouldWrite()` method.